src/main/scala/mesosphere/marathon/state/Migration.scala

```diff
@@ -371,21 +374,47 @@ class MigrationTo0_16(groupRepository: GroupRepository, appRepository: AppReposi
 }
 
 /**
-  * Removes all deployment version nodes from ZK
+  * Implements the following migration logic:
+  * * Removes all deployment version nodes from ZK
+  * * Adds calculated MarathonTaskStatus to stored tasks
   */
-class MigrationTo1_2(deploymentRepository: DeploymentRepository) {
+class MigrationTo1_2(deploymentRepository: DeploymentRepository, taskRepository: TaskRepository) {
   private[this] val log = LoggerFactory.getLogger(getClass)
 
   def migrate(): Future[Unit] = {
     log.info("Start 1.2 migration")
 
-    deploymentRepository.store.names().map(_.filter(deploymentRepository.isVersionKey)).flatMap { versionNodes =>
-      versionNodes.foldLeft(Future.successful(())) { (future, versionNode) =>
-        future.flatMap { _ =>
-          deploymentRepository.store.expunge(versionNode).map(_ => ())
+    val deploymentMigrationFuture = deploymentRepository.store.names()
+      .map(_.filter(deploymentRepository.isVersionKey)).flatMap { versionNodes =>
+        versionNodes.foldLeft(Future.successful(())) { (future, versionNode) =>
+          future.flatMap { _ =>
+            deploymentRepository.store.expunge(versionNode).map(_ => ())
+          }
         }
       }
-    }.flatMap { Unit =>
+
+    def loadAndMigrateTasks(id: String): Future[MarathonTask] = {
+      taskRepository.task(id).flatMap {
+        case Some(entity) =>
+          val updatedEntity = entity.toBuilder
+            .setMarathonTaskStatus(MarathonTaskStatusSerializer.toProto(MarathonTaskStatus(entity.getStatus)))
+            .build()
+          taskRepository.store(updatedEntity)
```

Comment:
Thinking about this, you should skip entities that already have a marathonTaskStatus. In case a migration attempt fails and is re-run, some of these might have been migrated already. That's not necessarily nice, but it's the way it is handled until now.